packages/@aws-cdk/aws-appmesh/lib/shared-interfaces.ts

```diff
@@ -194,3 +196,84 @@ class FileAccessLog extends AccessLog {
   }
 }
 
+/**
+ * Represents the properties needed to define backend defaults
+ */
+export interface BackendDefaults {
+  /**
+   * Client policy for backend defaults
+   *
+   * @default none
+   */
+  readonly clientPolicy?: ClientPolicy;
+}
+
+/**
+ * Represents the properties needed to define a Virtual Service backend
+ */
+export interface VirtualServiceBackendOptions {
+  /**
+   * The Virtual Service this backend points to
+   */
+  readonly virtualService: IVirtualService;
+
+  /**
+   * Client policy for the backend
+   *
+   * @default none
+   */
+  readonly clientPolicy?: ClientPolicy;
+}
+
+/**
+ * Properties for a backend
+ */
+export interface BackendConfig {
+  /**
+   * Config for a Virtual Service backend
+   */
+  readonly virtualServiceBackend: CfnVirtualNode.BackendProperty;
+}
+
+
+/**
+ * Contains static factory methods to create backends
+ */
+export abstract class Backend {
+  /**
+   * Construct a Virtual Service backend
+   */
+  public static virtualService(props: VirtualServiceBackendOptions): Backend {
+    return new VirtualServiceBackend(props.virtualService, props.clientPolicy);
+  }
```

Comment:
How about we add `virtualService` as the first positional argument to this method, and remove it from `VirtualServiceBackendOptions`?

The reason I ask is because, right now, you have to do:

```ts
Backend.virtualService({ virtualService: virtualService })
```

That's a lot of `virtualService`!

Instead, if we could do `Backend.virtualService(virtualService, { optionalPropsHere: false } )`, it seems a lot more natural.

Thoughts on this idea?

Reply:
A reference to the Virtual Service is a [required parameter](https://docs.aws.amazon.com/app-mesh/latest/APIReference/API_VirtualServiceBackend.html) so IMO this is a smart and safe simplification. done